configure.ac

```diff
@@ -6686,11 +6686,10 @@ int main(void)
 ]])],
 [ac_cv_strftime_c99_support=yes],
 [ac_cv_strftime_c99_support=no],
-[ac_cv_strftime_c99_support=no])])
-if test "$ac_cv_strftime_c99_support" = yes
+[ac_cv_strftime_c99_support=yes])])
+if test "$ac_cv_strftime_c99_support" = no
 then
-  AC_DEFINE([Py_STRFTIME_C99_SUPPORT], [1],
-  [Define if C99-specific strftime specifiers are supported.])
+  AC_MSG_ERROR([Python requires C99-compatible strftime specifiers])
 fi
```

Comment:
Let's just simplify this to adding the `AC_MSG_ERROR` in the _action-if-false_ clause of `AC_RUN_IFELSE`. (I think I got the suggestion right; please double check)

```suggestion
[],
[AC_MSG_ERROR([Python requires C99-compatible strftime specifiers])])])
```

Reply:
Great! I think I tried that, but don't recall what went wrong. I'll add the patch and test it.

Reply:
Ah okay; this patch works fine, but how does it interact with the `AC_CACHE_CHECK([whether C99-compatible strftime specifiers are supported], [ac_cv_strftime_c99_support], [ ...` bit above?

Reply:
We can drop that too, though I don't know if the caching matters?

```diff
diff --git a/configure.ac b/configure.ac
index ccb0549dc33..cad36ba1011 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6665,7 +6665,6 @@ then
   [Define if year with century should be normalized for strftime.])
 fi
 
-AC_CACHE_CHECK([whether C99-compatible strftime specifiers are supported], [ac_cv_strftime_c99_support], [
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <time.h>
 #include <string.h>
@@ -6685,7 +6684,7 @@ int main(void)
 }
 ]])],
 [],
-[AC_MSG_ERROR([Python requires C99-compatible strftime specifiers])])])
+[AC_MSG_ERROR([Python requires C99-compatible strftime specifiers])])
 
 dnl check for ncursesw/ncurses and panelw/panel
 dnl NOTE: old curses is not detected.
```

Reply:
So.. removing the caching does have an effect, it looks like it broke the WASI CI build

Caching can be used for cross-compiles discussed in https://www.gnu.org/software/autoconf/manual/autoconf-2.63/html_node/Runtime.html

> To configure for cross-compiling you can also choose a value for those parameters based on the canonical system name (see [Manual Configuration](https://www.gnu.org/software/autoconf/manual/autoconf-2.63/html_node/Manual-Configuration.html#Manual-Configuration)). Alternatively, set up a test results cache file with the correct values for the host system (see [Caching Results](https://www.gnu.org/software/autoconf/manual/autoconf-2.63/html_node/Caching-Results.html#Caching-Results)).

I'll see if adding an empty `action-if-cross-compiling` block changes the behavior

Reply:
https://github.com/python/cpython/pull/128106/commits/7928672470c7d606d5554d22f3b3a60603966d95 doesn't change the `configure` file, so it shouldn't have any affect. 

I think we'll need to skip this explicitly when cross-compiling

Reply:
The skip would look like this
```diff
diff --git a/configure b/configure
index 31af8b060c0..4ee003e0b21 100755
--- a/configure
+++ b/configure
@@ -26395,6 +26395,9 @@ printf "%s\n" "#define Py_NORMALIZE_CENTURY 1" >>confdefs.h
 
 fi
 
+if test "x$cross_compiling" = xyes
+then :
+
 if test "$cross_compiling" = yes
 then :
   { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
@@ -26433,6 +26436,7 @@ rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
 
+fi
 
 have_curses=no
 have_panel=no
diff --git a/configure.ac b/configure.ac
index cad36ba1011..0160e9d4c7e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6665,6 +6665,7 @@ then
   [Define if year with century should be normalized for strftime.])
 fi
 
+AS_VAR_IF([cross_compiling], [yes], [
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <time.h>
 #include <string.h>
@@ -6684,7 +6685,7 @@ int main(void)
 }
 ]])],
 [],
-[AC_MSG_ERROR([Python requires C99-compatible strftime specifiers])])
+[AC_MSG_ERROR([Python requires C99-compatible strftime specifiers])])])
 
 dnl check for ncursesw/ncurses and panelw/panel
 dnl NOTE: old curses is not detected.
 ```
 
 Curious for your thoughts. Retaining the cache seems most consistent with other uses of `AC_RUN_IFELSE` but I'm not sure if it's intentionally setup for cross-compiling purposes.